File: longest-common-prefix.cc
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
  string longestCommonPrefix(vector<string> &strs) {
    if (strs.empty()) return string();
    if (strs.size() == 1) return strs[0];
    string ans = commonPrefix(strs[0], strs[1]);
    for (int i = 2; i < strs.size(); i++)
      ans = commonPrefix(ans, strs[i]);
    return ans;
  }

  string commonPrefix(const string& a, const string& b) {
    int i = 0;
    while (i < a.size() && i < b.size() && a[i] == b[i])
      i++;
    return a.substr(0, i);
  }
};

File: longest-common-prefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return "";
        String ans = strs[0];
        for (int i = 1; i < strs.length; i++) {
            int max_len = 0;
            while (max_len < ans.length() && max_len < strs[i].length() &&
                    ans.charAt(max_len) == strs[i].charAt(max_len))
                max_len++;
            ans = ans.substring(0, max_len);
        }
        return ans;
    }
}
